User filter

vinhtruong94 opened the pull request:
Added and modified addressVerified for UserFilter, added some fields …
…for validation in IndexUserRequest, added scope method scopeWithRole for user role in User model, added getUsers for userResource.svc.js to get users from different filters, addedhtml buttons and html select options for user Filter in html file,added search functions for filters for userSearchBar js based on motionSearchBar ( motion Filter)

psaunders commented on app/Filters/UserFilter.php:
```diff
@@ -38,6 +38,11 @@ public function allNames($name = '')
                            ->orWhere('last_name', 'like', "%$name%");
     }
 
+    public function role($role = '')
```
There is already a "roles" query below, you need to pass an array like motion statuses

psaunders commented on app/Http/Requests/User/IndexUserRequest.php:
```diff
@@ -40,7 +40,7 @@ public function rules()
             'allNames'                => 'string',
             'page'                    => 'integer',
             'limit'                   => 'integer',
-            'roles'                   => 'array',
```
It should take roles, people can have 1:m roles

psaunders commented on app/User.php:
```diff
@@ -524,6 +524,13 @@ public function scopeDoesntHaveRoles($query, $role)
         });
     }
 
+    public function scopeWithRoles($query, $role)
```
It's just below it, why all the duplication. Just query with an array

psaunders commented on public/app/components/userSidebar/search/userSearchbar.tpl.html:
```diff
@@ -12,11 +12,39 @@
+        <md-menu-content width="4" class="md-body-1">
+
+            <md-select placeholder="Filter by Role" ng-model="search._role.filter" ng-change="search.searchSpecific()" aria-label="filter users by roles" layout-padding>
+              <md-option ng-value="search._role.filters[0].query">{{search._role.filters[0].name}}</md-option>
```
Should be a loop so changing it one place changes it here

psaunders commented on public/app/components/userSidebar/search/userSearchbar.tpl.html:
```diff
@@ -12,11 +12,39 @@
+            </md-select> 
+            
+            <md-select placeholder="Filter by Address" ng-model="search._addressVerified.filter" ng-change="search.searchSpecific()" aria-label="filter users by address" layout-padding>
+              <md-option ng-value="search._addressVerified.filters[0].query">{{search._addressVerified.filters[0].name}}</md-option>
```
What's with the underscores?

vinhtruong94 commented:
Undo redundant back-end codes. 
Added back roles validation in IndexUserRequest.
Tight up html to loop repeat instead of multi underscores.
Added front end js query for roles.
Tested on local front end, all features and buttons are functioned.
Please review the changes, you can pull to review new functions.
